aiida/orm/entities.py

```diff
@@ -163,6 +163,15 @@ def all(self):
         """
         return [_[0] for _ in self.query().all()]
 
+    def reset(self):
+        """
+        Remove lazily constructed collection instance.
+
+        Removes lazily constructed collection instance from LazyStore (constructed again upon next request).
+        Useful to clear any internal caches of the collection.
+        """
+        self._COLLECTIONS.pop((self.entity_type, self.backend))
```

Comment:
Won't this raise if the `(self.entity_type, self.backend)` tuple does not exist?

Reply:
It should definitely have tests ;-) 
If you agree to add this functionality, I'm happy to add some - just let me know.

Reply:
It feels like it is quite important functionality and with potentially big effects to just include in this PR. Would it be possible to just use the specific solution instead for now on the `User` collection. Just a `User.objects.set_default` or `User.objects.reset_default` even. I would feel a bit more comfortable with a more limited scope as part of this bigger PR that has really a completely different aim

Reply:
Ok, will do!

Reply:
I'm offline for a while now (will fix this PR over the weekend)